File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/VeremVeriSetiFormViewModel.cs
//$A6A7725D
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class VeremVeriSetiServiceController
    {
        partial void PreScript_VeremVeriSetiForm(VeremVeriSetiFormViewModel viewModel, VeremVeriSeti veremVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._VeremVeriSeti.EpisodeAction = episodeAction;
                viewModel._VeremVeriSeti.SubEpisode = episodeAction.SubEpisode;

            }
            ContextToViewModel(viewModel, objectContext);
        }
    }
}

namespace Core.Models
{
    public partial class VeremVeriSetiFormViewModel: BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._VeremVeriSeti);

            if (this.VeremHastalikTutumYeriGridList != null)   // Ek nesneler için
            {
                foreach (var item in this.VeremHastalikTutumYeriGridList)
                {
                    var veremHastalikTutumYeri = (VeremHastalikTutumYeri)objectContext.AddObject(item);
                    veremHastalikTutumYeri.VeremVeriSeti = this._VeremVeriSeti;
                }
            }

            if (this.VeremIlacBilgisiGridList != null)
            {
                foreach (var item in this.VeremIlacBilgisiGridList)
                {
                    var veremIlacBilgisi = (VeremIlacBilgisi)objectContext.AddObject(item);
                    veremIlacBilgisi.VeremVeriSeti = this._VeremVeriSeti;
                }
            }

            if (this.VeremKlinikOrnegiGridList != null)
            {
                foreach (var item in this.VeremKlinikOrnegiGridList)
                {
                    var veremKlinikOrnegi = (VeremKlinikOrnegi)objectContext.AddObject(item);
                    veremKlinikOrnegi.VeremVeriSeti = this._VeremVeriSeti;
                }
            }

            if (this.VeremTedaviSonucBilgisiGridList != null)
            {
                foreach (var item in this.VeremTedaviSonucBilgisiGridList)
                {
                    var veremTedaviSonucBilgisi = (VeremTedaviSonucBilgisi)objectContext.AddObject(item);
                    veremTedaviSonucBilgisi.VeremVeriSeti = this._VeremVeriSeti;
                }
            }



            var veremVeriSeti = (VeremVeriSeti)objectContext.GetLoadedObject(this._VeremVeriSeti.ObjectID);


            new SendToENabiz(objectContext, veremVeriSeti.SubEpisode, veremVeriSeti.ObjectID, veremVeriSeti.ObjectDef.Name, "250", Common.RecTime());
        }
    }
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/VeremVeriSetiFormViewModel.ts
//$458511E3
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { VeremVeriSeti } from "NebulaClient/Model/AtlasClientModel";
import { VeremTedaviSonucBilgisi } from "NebulaClient/Model/AtlasClientModel";
import { VeremKlinikOrnegi } from "NebulaClient/Model/AtlasClientModel";
import { VeremIlacBilgisi } from "NebulaClient/Model/AtlasClientModel";
import { VeremHastalikTutumYeri } from "NebulaClient/Model/AtlasClientModel";
import { SKRSYaymaSonucu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSVeremOlguTanimi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSVeremHastasiTedaviYontemi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSKulturSonucu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSHastaninTedaviyeUyumu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSDGTUygulamasiniYapanKisi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSDGTUygulamaYeri } from "NebulaClient/Model/AtlasClientModel";
import { SKRSVeremTedaviSonucu } from "NebulaClient/Model/AtlasClientModel";
import { SKRSVeremHastasiKlinikOrnegi } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIlacAdiVerem } from "NebulaClient/Model/AtlasClientModel";
import { SKRSIlacDirenciVerem } from "NebulaClient/Model/AtlasClientModel";
import { SKRSVeremHastaligininTutulumYeri } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class VeremVeriSetiFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.VeremVeriSetiFormViewModel, Core';
    @Type(() => VeremVeriSeti)
    public _VeremVeriSeti: VeremVeriSeti = new VeremVeriSeti();
    @Type(() => VeremTedaviSonucBilgisi)
    public VeremTedaviSonucBilgisiGridList: Array<VeremTedaviSonucBilgisi> = new Array<VeremTedaviSonucBilgisi>();
    @Type(() => VeremKlinikOrnegi)
    public VeremKlinikOrnegiGridList: Array<VeremKlinikOrnegi> = new Array<VeremKlinikOrnegi>();
    @Type(() => VeremIlacBilgisi)
    public VeremIlacBilgisiGridList: Array<VeremIlacBilgisi> = new Array<VeremIlacBilgisi>();
    @Type(() => VeremHastalikTutumYeri)
    public VeremHastalikTutumYeriGridList: Array<VeremHastalikTutumYeri> = new Array<VeremHastalikTutumYeri>();
    @Type(() => SKRSYaymaSonucu)
    public SKRSYaymaSonucus: Array<SKRSYaymaSonucu> = new Array<SKRSYaymaSonucu>();
    @Type(() => SKRSVeremOlguTanimi)
    public SKRSVeremOlguTanimis: Array<SKRSVeremOlguTanimi> = new Array<SKRSVeremOlguTanimi>();
    @Type(() => SKRSVeremHastasiTedaviYontemi)
    public SKRSVeremHastasiTedaviYontemis: Array<SKRSVeremHastasiTedaviYontemi> = new Array<SKRSVeremHastasiTedaviYontemi>();
    @Type(() => SKRSKulturSonucu)
    public SKRSKulturSonucus: Array<SKRSKulturSonucu> = new Array<SKRSKulturSonucu>();
    @Type(() => SKRSHastaninTedaviyeUyumu)
    public SKRSHastaninTedaviyeUyumus: Array<SKRSHastaninTedaviyeUyumu> = new Array<SKRSHastaninTedaviyeUyumu>();
    @Type(() => SKRSDGTUygulamasiniYapanKisi)
    public SKRSDGTUygulamasiniYapanKisis: Array<SKRSDGTUygulamasiniYapanKisi> = new Array<SKRSDGTUygulamasiniYapanKisi>();
    @Type(() => SKRSDGTUygulamaYeri)
    public SKRSDGTUygulamaYeris: Array<SKRSDGTUygulamaYeri> = new Array<SKRSDGTUygulamaYeri>();
    @Type(() => SKRSVeremTedaviSonucu)
    public SKRSVeremTedaviSonucus: Array<SKRSVeremTedaviSonucu> = new Array<SKRSVeremTedaviSonucu>();
    @Type(() => SKRSVeremHastasiKlinikOrnegi)
    public SKRSVeremHastasiKlinikOrnegis: Array<SKRSVeremHastasiKlinikOrnegi> = new Array<SKRSVeremHastasiKlinikOrnegi>();
    @Type(() => SKRSIlacAdiVerem)
    public SKRSIlacAdiVerems: Array<SKRSIlacAdiVerem> = new Array<SKRSIlacAdiVerem>();
    @Type(() => SKRSIlacDirenciVerem)
    public SKRSIlacDirenciVerems: Array<SKRSIlacDirenciVerem> = new Array<SKRSIlacDirenciVerem>();
    @Type(() => SKRSVeremHastaligininTutulumYeri)
    public SKRSVeremHastaligininTutulumYeris: Array<SKRSVeremHastaligininTutulumYeri> = new Array<SKRSVeremHastaligininTutulumYeri>();
}
